src/models/DRBResult.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+  DRBWork,
+  Edition,
+  Author,
+  Agent,
+  EditionLink,
+} from "../types/drbTypes"
+import { SOURCE_PARAM, DRB_BASE_URL } from "../config/constants"
+import { readOnlineMediaTypes, downloadMediaTypes } from "../utils/drbUtils"
+/**
+ * The DRBResult class contains the data and getter functions
+ * for a single DRB search result entity.
+ *
+ * DRB Results returned from the API are mapped and initialized into
+ * DRBResult objects in the DRBContainer component.
+ */
+export default class DRBResult {
+  id: string
+  title: string
+  editions?: Edition[]
+  authors?: Author[] | Agent[]
+
+  constructor(work: DRBWork) {
+    this.id = work.uuid
+    this.title = work.title
+    this.editions = work.editions
+    this.authors = this.getAuthorsFromWork(work)
+  }
+
+  get url(): string {
+    return `${DRB_BASE_URL}/work/${this.id}${SOURCE_PARAM}`
+  }
+
+  get selectedEdition(): Edition {
```

Comment:
Where does `this.selectedEdition` get assigned? there are checks to verify that `this.selectedEdition.items` is not null but I don't see where it's assigned.